nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -390,7 +401,8 @@ def configure_optimizers(self):
             assert self._trainer.max_steps is not None, "'max_steps' is missing in trainer config."
             if hasattr(self._cfg.optim, 'sched'):
                 sched_config = self._cfg.optim.sched
-                sched_config['max_steps'] = self._trainer.max_steps
+                if 'max_steps' not in sched_config:
+                    sched_config['max_steps'] = self._trainer.max_steps
```

Comment:
Looks fine but can you change the name of `max_steps`? Something to help describe the difference between your `max_steps` and `trainer.max_steps`

Reply:
maybe decay_steps?